pkg/cri/config/config.go

```diff
@@ -329,6 +329,8 @@ type PluginConfig struct {
 	// The string is in the golang duration format, see:
 	//   https://golang.org/pkg/time/#ParseDuration
 	ImagePullProgressTimeout string `toml:"image_pull_progress_timeout" json:"imagePullProgressTimeout"`
+	// ImagePullWithTransferService specifies whether to pull image using transfer service otherwise it uses client library
+	ImagePullWithTransferService bool `toml:"image_pull_with_transfer_service" json:"imagePullWithTransferService"`
```

Comment:
@dmcgowan wdyt what he has here uses transfer with a new default config but if not "set" to true transfer would be disabled

as an alternative we could make this a disable bool with default off.. giving someone the option of disabling it if they have a problem.. 

 do you think we need to cover our bets with a default off for 2.0 for moving to transfer?

Reply:
> @dmcgowan wdyt what he has here uses transfer with a new default config but if not "set" to true transfer would be disabled

Yeah, now I set the default to true is more for test purpose, I am thinking of default it to off. 
I was following Derek's approach on ctr using transfer service, i.e. has an [local flag ](https://github.com/containerd/containerd/blob/main/cmd/ctr/commands/images/pull.go#LL79C20-L79C20) default to off, but users can optionally turn it on. 
For CRI, should we follow the same? I am open for suggestion.

Reply:
I think keeping it default to on is good. Ideally we have time before 2.0 to just remove the option all together but we don't need to rush that.

Reply:
right and if default should be on.. I'd prefer unset to mean on.. thus.. let's use a disable bool instead of an enable bool and as you say we can change our minds if it's not stable

Reply:
Switched to `DisableImagePullWithTransferService`, and default to `false`